### PR TITLE
Wallet list accounts

### PR DESCRIPTION
**Pull request by qustavo**
#### Pull Request Checklist

- [x] PR is opened against correct version branch.
- [x] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [x] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

### CODE REVIEW

File: walletkit_client.go [hunk 1]
```diff
@@ -386,3 +392,25 @@ func (m *walletKitClient) BumpFee(ctx context.Context, op wire.OutPoint,
...
+// ListAccounts retrieves all accounts belonging to the wallet by default.
+// Optional name and addressType can be provided to filter through all of the
+// wallet accounts and return only those matching.
+func (m *walletKitClient) ListAccounts(ctx context.Context, name string,
+	addressType walletrpc.AddressType) ([]*walletrpc.Account, error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, m.timeout)
+	defer cancel()
+
+	resp, err := m.client.ListAccounts(rpcCtx,
```
Comment (guggero) [inline walletkit_client.go:405]: nit: break before `rpcCtx`.

**Review by guggero — APPROVED** [review]
Nice, thanks! LGTM :tada:

File: walletkit_client.go (hunk 1 above)
Comment (guggero) [reply walletkit_client.go]: Ah yes, of course.

Comment (qustavo) [reply walletkit_client.go]: fixed your comment and added `.WithMacaroonAuth`

**Review by carlaKC — APPROVED** [review]
Thanks for the PR!

**Pull request merged**